### PR TITLE
[WIP] refactor: restructure components

### PR DESCRIPTION
**Pull request by geekplux**

### CODE REVIEW

**Comment by yihong0618** [comment]
Can we keep the old eslint format like `single quote`

**Comment by geekplux** [comment]
> Can we keep the old eslint format like `single quote`

absolutely, and it is..please review carefully.

File: src/hooks/useHover.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { useState } from 'react';
+
+// Hooks
+const useHover = () => {
+  const [hovered, setHovered] = useState();
+  const [timer, setTimer] = useState();
+
+  const eventHandlers = {
+    onMouseOver() {
+      setTimer(setTimeout(() => setHovered(true), 700));
```
Comment (yihong0618) [inline src/hooks/useHover.js:10]: Maybe we can make this 700 config too. But I don't know if it is worth.~

**Review by yihong0618 — APPROVED** [review]
Really wonderful work~
Thanks a lot~
I think we can  cross out one of the todo's , haha

File: src/hooks/useHover.js (hunk 1 above)
Comment (geekplux) [reply src/hooks/useHover.js:10]: haha, premature optimization is the root of all evil. Currently, we should modularize all components and find out which part of logic is independent or dependent, and abstract them respectively.

**Comment by geekplux** [comment]
Now, although all the components have been stripped out, the coupling of the page is very high. Especially **the first bunch of global variables** in the `Index page`, this part needs to be abstracted out. In addition, the encapsulation of some functions is a bit unnecessary and can be removed as appropriate.

**Comment by yihong0618** [comment]
vercel ci seems failed

**Comment by geekplux** [comment]
> vercel ci seems failed

I was wondering what the error is, could you please share it?

**Comment by yihong0618** [comment]
> > vercel ci seems failed
> 
> I was wondering what the error is, could you please share it?

image

**Review by yihong0618 — COMMENTED** [review]
coool

**Comment by geekplux** [comment]
It's ready to merge. I think the next step is to refactor the interaction login in Index page, which is a long term task with no rush.

**Pull request merged**